### PR TITLE
[folly]: fix lz4 Snappy deps

### PR DESCRIPTION
**Pull request by jiayuehua**
**Describe the pull request**
 fix folly port's LZ4 Snappy deps
- #### What does your PR fix?
  Fixes folly LZ4 Snappy deps #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Review by Adela0814 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

File: ports/folly/fix-deps.patch [hunk 1]
```diff
@@ -21,6 +21,12 @@ index 1689f9a..1d3295a 100644
 +if (NOT @CMAKE_DISABLE_FIND_PACKAGE_unofficial-sodium@)
 +    find_dependency(unofficial-sodium CONFIG)
 +endif()
++if (NOT @CMAKE_DISABLE_FIND_PACKAGE_Snappy@)
```
Comment (dg0yt) [inline ports/folly/fix-deps.patch:24]: This `@...@` should have been quoted.

File: ports/folly/fix-deps.patch [hunk 2]
```diff
@@ -21,6 +21,12 @@ index 1689f9a..1d3295a 100644
...
++	find_dependency(Snappy CONFIG)
```
Comment (dg0yt) [inline ports/folly/fix-deps.patch:25]: Inconsistent intentation.

File: ports/folly/fix-deps.patch [hunk 3]
```diff
@@ -105,8 +111,20 @@ index 4b78e9f..6228bf9 100644
...
+   list(APPEND FOLLY_LINK_LIBRARIES ${LIBLZMA_LIBRARIES})
+ endif()
+ 
+-find_package(LZ4 MODULE)
+-set(FOLLY_HAVE_LIBLZ4 ${LZ4_FOUND})
+-if (LZ4_FOUND)
+-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LZ4_INCLUDE_DIR})
+-  list(APPEND FOLLY_LINK_LIBRARIES ${LZ4_LIBRARY})
++if (NOT CMAKE_DISABLE_FIND_PACKAGE_LZ4)
++  find_package(lz4 CONFIG)
```
Comment (dg0yt) [inline ports/folly/fix-deps.patch:124]: Is this required now, or can it break silently?

File: ports/folly/fix-deps.patch [hunk 4]
```diff
@@ -125,7 +143,19 @@ index 4b78e9f..6228bf9 100644
...
  endif()
  
- find_package(Snappy MODULE)
+-find_package(Snappy MODULE)
+-set(FOLLY_HAVE_LIBSNAPPY ${SNAPPY_FOUND})
+-if (SNAPPY_FOUND)
+-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${SNAPPY_INCLUDE_DIR})
+-  list(APPEND FOLLY_LINK_LIBRARIES ${SNAPPY_LIBRARY})
++if (NOT CMAKE_DISABLE_FIND_PACKAGE_Snappy)
++    find_package(Snappy CONFIG)
```
Comment (dg0yt) [inline ports/folly/fix-deps.patch:152]: Is this required now, or can it break silently?

**Review by dg0yt — COMMENTED** [review]
Was this PR really ready?

**Pull request merged**